### PR TITLE
Add integration tests for ChatStateService and Chat states.

### PR DESCRIPTION
**Pull request by Deletrius366**

### CODE REVIEW

**Comment by Deletrius366** [comment]
For now, AddFilledTodoTest is broken because of the bug in AddStartDateState and AddStartEndState. I've created corresponding issues.

**Comment by Garuda-1** [comment]
> For now, AddFilledTodoTest is broken because of the bug in AddStartDateState and AddStartEndState. I've created corresponding issues.

Fixed now, you can pull changes from the `dev` branch.

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -6,6 +6,8 @@ spring:
     url: ${DATASOURCE_URL}
     username: ${DATASOURCE_USERNAME}
     password: ${DATASOURCE_PASSWORD}
+  main:
+    allow-bean-definition-overriding: true
```
Comment (Garuda-1) [inline src/main/resources/application.yaml:10]: Could you explain what does this property change?
If you want to use it in tests only, it'd be better to either set in `@SpringBootTest(properties = ...)` or create `application.yaml` in `test/main/resources` and set it there.

**Review by Garuda-1 — CHANGES_REQUESTED** [review]
Looks awesome as a beginning. However, we'd need more scenarios, I'll share my ideas a bit later.

File: src/main/resources/application.yaml (hunk 1 above)
Comment (Deletrius366) [reply src/main/resources/application.yaml:10]: we override `taskFetchingService` in  `ServiceTestConfiguration` and `SpringBootTest` doesn't like it without special option. Anyway, I realized that I do not need config, so it should be removed.

File: src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+    @BeforeEach
+    public void setUp() {
+        now = LocalDateTime.of(2000, 1, 1, 0, 0, 0);
+        processMessage("/start");
+    }
+
+    @Test
+    @DisplayName(value = "Start bot")
+    public void StartTest() {
```
Comment (Garuda-1) [inline src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java]: Why not lowerCamelCase?

File: src/main/resources/application.yaml (hunk 1 above)
Comment (Deletrius366) [reply src/main/resources/application.yaml:10]: Interesting issue - with removing `ServiceTestConfiguration`  if I run all tests (run ChatStateServiceTest) - tests pass. But if I run one test (e.g. StartTest) - it fails. With `ServiceTestConfiguration` and `allow-bean-definition-overriding: true` - all tests pass. For me, tests should pass even without `ServiceTestConfiguration`, but no :(

File: src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java (hunk 2 above)
Comment (Deletrius366) [reply src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java]: Fixed

File: src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    @Test
+    @DisplayName(value = "Start bot")
+    public void startTest() {
+        checkChatState(ChatStateEnum.BASE_STATE);
+    }
+
+    @Test
+    @DisplayName(value = "Add todo")
```
Comment (Garuda-1) [inline src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java:58]: `todo` -> `TODO`
Same below

File: src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        processMessage("/create_new_todo");
+        checkChatState(ChatStateEnum.ADD_NAME_STATE);
+
+        processMessage("First Todo");
+        checkTodo("First Todo");
+        checkChatState(ChatStateEnum.EDIT_TODO_STATE);
+    }
+
+    @Test
+    @DisplayName(value = "Add 2 todo")
```
Comment (Garuda-1) [inline src/test/java/ru/itmo/sd/deadliner2bot/service/ChatStateServiceTest.java:69]: `todo` -> `TODOs`

**Review by Garuda-1 — CHANGES_REQUESTED** [review]
I'll check the tests logic tomorrow

**Review by Garuda-1 — APPROVED** [review]

**Pull request merged**